doc/lightning-close.7.md

```diff
@@ -4,7 +4,7 @@ lightning-close -- Command for closing channels with direct peers
 SYNOPSIS
 --------
 
-**close** *id* [*unilateraltimeout*] [*destination*] [*fee_negotiation_step*] [*wrong_funding*] [*force_lease_closed*] [\*feerange\*]
+**close** *id* [*unilateraltimeout*] [*destination*] [*fee_negotiation_step*] [*wrong_funding_txid*] [*force_lease_closed*] [\*feerange\*]
```

Comment:
nit: this break the code https://github.com/ElementsProject/lightning/blob/e70729b04befe44d603370b96fbb10dc00bd342d/lightningd/closing_control.c#L627

Maybe we have the json schema wrong?

Reply:
The correct way of doing this is accept both for a deprecation period, gating the old one on `deprecated_apis == true`, so it won't show if we run without it, and after 6 months ~= 3 releases we can remove the old one and any merge logic on the two.

Reply:
@cdecker So you agree this way is better and I just need to add deprecation code?

Reply:
I want to clarify my review here, the nit is regarding the docs changes because what you are changing is not true in the code.

However, the suffix `txid` will make more clear the command line options type, and this is good.

Reply:
I do agree that the `txid` suffix makes it much clearer. It's just that renaming things in the public interface sadly comes with the deprecation window requirements, which is why we usually refrain from addressing these types of issues.

Would absolutely accept if this had the necessary deprecation window added.

Reply:
Hmm... but it's actually outpoint, not txid. Maybe proper name would be `wrong_funding_outpoint`?

Reply:
Also true, man, naming is hard ^^

Reply:
ok, why not left as is :) and adding a `FIXME: adding a new cool name` should fit in the requirements of when use a `FIXME`

Reply:
> ok, why not left as is :) and adding a `FIXME: adding a new cool name` should fit in the requirements of when use a `FIXME`

Renaming and adding deprecation window is not so hard, don't see point in adding some FIXME comment here.

Reply:
>Renaming and adding deprecation window is not so hard, don't see point in adding some FIXME comment here.

From ho provide the API yes it is easy, from ho use the API is a pain any way. The important think is that we keep the convention by naming along all the API.

My vote for the naming is the same, summurized:

- `wrong_funding_txid`: could work
- `wrong_funding_outpoint`: good luck if you need to type this string on the keyboard, long name is painful if you think accessibility for people that have some disabilities.
- `wrong_funding`: generic name that include both of previous,  so if there is no agreement I preferer stay with the generic one

Reply:
Probabaly you're right. Then alternative is to rename `wrong_funding_txid` to `wrong_funding` in DESCRIPTION in manpage and also fix type in `doc/schemas/close.request.json`, which is also wrong (should be `outpoint` not `txid`).

Reply:
I think the docs fix is a very improvement, and also if you think that the name is not so clear (and it is not clear IMHO) you could al a small explanation in the docs

Reply:
It's explained good enough in DESCRIPTION part of manpage. Ok, will just rename it properly there and fix type in schema. That should not be controversial. And if decision is made to rename it, can be done later.